validator/client/src/test/java/tech/pegasys/teku/validator/client/EpochDutiesTest.java

```diff
@@ -108,4 +123,44 @@ public void shouldPerformDelayedDutiesInOrder() {
     inOrder.verify(scheduledDuties).performAggregation(ONE);
     inOrder.verifyNoMoreInteractions();
   }
+
+  @Test
+  void shouldRecalculateDuties() {
+    scheduledDutiesFuture.complete(scheduledDuties);
+
+    final ScheduledDuties newDuties = mock(ScheduledDuties.class);
+    final SafeFuture<ScheduledDuties> recalculatedDuties = new SafeFuture<>();
+    when(dutyLoader.loadDutiesForEpoch(EPOCH)).thenReturn(recalculatedDuties);
+
+    duties.recalculate();
+    verify(dutyLoader, times(2)).loadDutiesForEpoch(EPOCH);
+
+    // Should use new duties to perform actions
+    duties.onBlockProductionDue(ZERO);
```

Comment:
The EPOCH is set to 10, and we're asking for block production for slot 0. Won't we have no duties scheduled for that?

Reply:
They're mocks so it doesn't really matter and the `EpochDuties` is just there to proxy the events it gets through, delaying them to wait for loading if needed.